From 68e169a3cb3d65bda6fbb3f785d9c51b2d5e560c Mon Sep 17 00:00:00 2001
From: Yinhang Liu <yinhang.liu@intel.com>
Date: Mon, 20 Feb 2023 10:17:53 +0800
Subject: [PATCH 038/232] msdkvpp: correct the fixated caps for src pad

In src pad, fixate othercaps instead of the NULL result caps

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/3999>
---
 sys/msdk/gstmsdkvpp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sys/msdk/gstmsdkvpp.c b/sys/msdk/gstmsdkvpp.c
index 74d2a0855..fa837adb3 100644
--- a/sys/msdk/gstmsdkvpp.c
+++ b/sys/msdk/gstmsdkvpp.c
@@ -1453,7 +1453,7 @@ gst_msdkvpp_fixate_caps (GstBaseTransform * trans,
   gboolean *use_dmabuf;
 
   if (direction == GST_PAD_SRC) {
-    result = gst_caps_fixate (result);
+    result = gst_caps_fixate (othercaps);
     use_dmabuf = &thiz->use_sinkpad_dmabuf;
   } else {
     /*
-- 
2.40.1

